### ISSUE TITLE
kedro-datasets[pandas] install fails on windows

### ISSUE THREAD

**Issue by Galileo-Galilei** [kedro-org/kedro-plugins#402]
## Description

``pip install kedro-datasets[pandas]`` badly fails on Windows with an ugly stracktrace. 

## Context

I've been upgrading my old projects to finally use kedro-datasets. The recommended way to install ``pandas`` dependency (e.g. to read csv files with ``CSVDataset``) is to use ``pip install kedro-datasets[pandas]`` instead of ``pip install kedro-datasets pandas``. However the first command crashes badly while the second work, which goes against what is advised in the documentation. 

## Steps to Reproduce

Create a fresh conda env: 

```bash
conda create -n bug-kedro-pandas python=3.10 -y
conda activate bug-kedro-pandas
pip install kedro==0.18.14
pip install kedro-datasets
```
Now this fails (see stacktrace below): 
```bash
pip install kedro-datasets[pandas]
```

while this works: 

```bash
pip install pandas
```
## Expected Result

Install should work.

## Actual Result

```bash
(bug-kedro-pandas) C:\Users\>pip install kedro-datasets[pandas]
Requirement already satisfied: kedro-datasets[pandas] in c:\users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages (1.7.1)
Requirement already satisfied: kedro>=0.16 in c:\users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages (from kedro-datasets[pandas]) (0.18.14)
Requirement already satisfied: lazy-loader in c:\users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages (from kedro-datasets[pandas]) (0.3)
Collecting SQLAlchemy<3.0,>=1.4 (from kedro-datasets[pandas])
  Using cached SQLAlchemy-2.0.22-cp310-cp310-win_amd64.whl.metadata (9.7 kB)
Collecting deltalake>=0.10.0 (from kedro-datasets[pandas])
  Using cached deltalake-0.12.0-cp37-abi3-win_amd64.whl.metadata (5.1 kB)
Collecting lxml~=4.6 (from kedro-datasets[pandas])
  Using cached lxml-4.9.3-cp310-cp310-win_amd64.whl.metadata (3.9 kB)
Collecting openpyxl<4.0,>=3.0.6 (from kedro-datasets[pandas])
  Using cached openpyxl-3.1.2-py2.py3-none-any.whl (249 kB)
Collecting pandas-gbq<0.18.0,>=0.12.0 (from kedro-datasets[pandas])
  Using cached pandas_gbq-0.17.9-py2.py3-none-any.whl (25 kB)
Collecting pandas<3.0,>=1.3 (from kedro-datasets[pandas])
  Using cached pandas-2.1.1-cp310-cp310-win_amd64.whl.metadata (18 kB)
Collecting pyarrow>=6.0 (from kedro-datasets[pandas])
  Using cached pyarrow-13.0.0-cp310-cp310-win_amd64.whl.metadata (3.1 kB)
Collecting pyodbc~=4.0 (from kedro-datasets[pandas])
  Using cached pyodbc-4.0.39-cp310-cp310-win_amd64.whl (69 kB)
Collecting tables~=3.6.0 (from kedro-datasets[pandas])
  Using cached tables-3.6.1.tar.gz (4.6 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      C:\Users\...\AppData\Local\Temp\pip-install-b4ce1dsx\tables_5b8b01ba41804417b71ba4bb574b9d6a\setup.py:18: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
        import pkg_resources
      * Using Python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:34:57) [MSC v.1936 64 bit (AMD64)]
      * USE_PKGCONFIG: False
      * Found conda env: ``C:\Users\Yolan\mambaforge\envs\bug-kedro-pandas``
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\...\AppData\Local\Temp\pip-install-b4ce1dsx\tables_5b8b01ba41804417b71ba4bb574b9d6a\setup.py", line 634, in <module>
          libdir = compiler.has_function(package.target_function,
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\_distutils\ccompiler.py", line 893, in has_function
          objects = self.compile([fname], include_dirs=include_dirs)
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\_distutils\_msvccompiler.py", line 343, in compile
          self.initialize()
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\_distutils\_msvccompiler.py", line 253, in initialize
          vc_env = _get_vc_env(plat_spec)
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\msvc.py", line 233, in msvc14_get_vc_env
          return _msvc14_get_vc_env(plat_spec)
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\msvc.py", line 190, in _msvc14_get_vc_env
          raise distutils.errors.DistutilsPlatformError("Unable to find vcvarsall.bat")
      distutils.errors.DistutilsPlatformError: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

```
-- Separate them if you have more than one.
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used (`pip show kedro` or `kedro -V`): ``kedro==0.18.14``
* Kedro plugin and kedro plugin version used (`pip show kedro-airflow`): ``kedro-datasets==1.7.1``
* Python version used (`python -V`): Python 3.10.12
* Operating system and version: Windows 11

## Potential fix: 

The usual suspect is :     
https://github.com/kedro-org/kedro-plugins/blob/b44bf0e9db43d1564e50bbc85c7a61676c1ddbe4/kedro-datasets/setup.py#L37-L42

And I confirm this is actually what's breaking. Funnily install ``pip install tables`` installs "tables=3.9.1" correctly (but I don't have the local setup to test the dataset).

**Comment by astrojuanlu** [comment]
Thanks @Galileo-Galilei for reporting - so loosening the upper version cap on Windows should be enough then?

**Comment by Galileo-Galilei** [comment]
"It works on my machine" 😅 

Joke aside, I'll test with python 3.8 for tthe specific rule "tables~=3.6, <3.9; platform_system != 'Windows' and python_version<'3.9'". 

For next times, I guess it is a bad idea to pin an upper dependency because a new realease has broken something. This is a problem on their side, kedro is not responsible for fixing it. 

The main drawback is that when they release a fix (it seems 3.9.1), kedro has to relax its dependecies too, which add an extra burden to keep everything up to date (and spoiler : we don't manage too :)). We had a long discussion with @antonymilne back then about capping dependencies or not and it may be worth bringing it up to the table again.

**Comment by astrojuanlu** [comment]
> For next times, I guess it is a bad idea to pin an upper dependency because a new realease has broken something. This is a problem on their side, kedro is not responsible for fixing it.

I 💯 % agree, you don't have to convince me on this one :)
